service/interface.go

```
@@ -10,7 +10,7 @@ import (
 type Manager interface {
 	CreateUser(usr *domain.User) *domain.Error
 	UpdateUser(usr *domain.User) error
-	ListUser(limit int32, name string) []*domain.User
+	ListUser(int32, map[string]interface{}) []*domain.User
```

you need to return the error along with the user's list.

---

Done. Kindly Check